lib/containers/common.pm

```diff
@@ -25,7 +25,25 @@ use utils qw(zypper_call systemctl);
 use version_utils qw(is_sle is_leap is_microos is_opensuse is_jeos is_public_cloud);
 
 our @EXPORT = qw(install_podman_when_needed install_docker_when_needed allow_selected_insecure_registries clean_container_host
-  test_container_runtime test_container_image scc_apply_docker_image_credentials scc_restore_docker_image_credentials);
+  test_container_runtime test_container_image scc_apply_docker_image_credentials scc_restore_docker_image_credentials install_podman_centos install_docker_centos);
+
+sub install_podman_centos {
```

Comment:
I would say better to keep this inside `install_podman_when_needed`

Reply:
i tried but the conditions are not helping. So for simplicity it would be better to be in its own subroutine.

Reply:
I tend to agree with Anton, we could introduce a new variable `HOST_OS=centos` since `DISTRI` is still `sle`.

Reply:
Why not to avoid the overhead of extra variables' handling. from the design point of you the `install_podman_when_needed` should be unified of all the cases. But it is not. To handle installs in other distributions we have to introduce a couple of if..else and add new variables. i tried to apply open for extension/close for modification principle. So Instead, we can have dedicate subroutines for each case that we can schedule with the yaml scheduler. Then we do not need to touch things that work. And you do not need to know about extra variables but just what module do you need to schedule to run a test scenario. Does this make sense? i cant see why the introduction of a new variable will be more convenient in this case. What potential problems do you see?